### ISSUE TITLE
import mc

### ISSUE THREAD

**Issue by Elaineok** [KyleZheng1997/WCL#2]
There are no “mc” package in line5 of “imagenet.py” file.

![image](https://user-images.githubusercontent.com/46559128/144186831-e34e0e14-3d79-465a-9bf4-b99cc63ace24.png)

Can you help me solve this problem? Looking forward to your reply.

**Comment by mingkai-zheng** [comment]
Hello Jingting, the mc module is a caching library for speeding up the image io process, this is a special module that we used internally, so you can just simply delete the related code. 
As I have mentioned in README file, "You need to change the Dataset setting (dataset/imagenet.py), and Pytorch DDP setting (util/dist_init.py) for your own server environments."

**Comment by Elaineok** [comment]
Thank you for your reply！I have deleted the related code.

Can you provide a simple demo dataset and upload it to Baidu Netdisk? For example: contain a few images、train.txt.

 Looking forward to your reply.

**Comment by mingkai-zheng** [comment]
I think the most straightforward way is to replace the dataset with cifar10, you can simply follow the official [PyTorch tutorial](https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html). BTW, since the resolution of cifar10 images is relatively small (32x32), we generally need to replace the first 7x7 Conv of stride 2 with 3x3 Conv of stride 1 and also remove the first max pool operation for our backbone network.

**Comment by Elaineok** [comment]
It's a great suggestion. I want to have a try.

**Comment by Elaineok** [comment]
I tried, but found that the data loading part needs more changes. Can you provide a simple demo dataset？

Thank you very much.

**Comment by Elaineok** [comment]
I noticed that the data set and code need access permission, can you provide me with permission?
https://drive.google.com/file/d/1j2I1Lh9Dy7cHb6YO0PZ8HXDNewXrHO-j/view?usp=sharing
![image](https://user-images.githubusercontent.com/46559128/144344683-314fc2f3-b8e0-4974-99e2-8fadb1de2a2b.png)

**Comment by mingkai-zheng** [comment]
sorry for the late response, I have updated the permission, please try it again.

**Comment by jingtingxu369** [comment]
Since I cannot use distributed computing, i changed the code.

source code
![image](https://user-images.githubusercontent.com/46559128/144451502-94887398-59d3-488f-ab6b-1dfd6d21057a.png)
changed code
![image](https://user-images.githubusercontent.com/46559128/144451693-2f416dba-07dd-48d3-86b6-02d85d6ff162.png)

Is this correct? Looking forward to your reply.

**Comment by Elaineok** [comment]
Since I cannot access to distributed operations at will,  I have to change the code. At present, I am not very clear about what these sentence means.
![image](https://user-images.githubusercontent.com/46559128/144529845-7dcf8fc9-e4ec-45f2-afb1-533abe7bdaf6.png)

if feat1=（128，4096），128 represents batch_size，4096 represents feature_dimentation

 Looking forward to your reply.

**Comment by mingkai-zheng** [comment]
The shape of feat1   is (batch_size_per_gpu, dim)

The shape of other1 is (batch_size_per_gpu * (world_size - 1), dim), 

The concat_other_gather function returns the features from other GPUs.

torch.cat([feat1, feat2]) will concat all the features in current GPU.

torch.cat([feat1, feat2, other1, other2]) will concat all the features across all the GPUs.

If you only have a single GPU, remove line 55 and 56 and change line 59 to prob = torch.cat([feat1, feat2]) @ torch.cat([feat1, feat2]).T / t

**Comment by a139122679** [comment]
I want to know how to delete the code of this MC module, I tried to delete but failed to run, or can you provide an MC package, thank you.

**Comment by mingkai-zheng** [comment]
> I want to know how to delete the code of this MC module, I tried to delete but failed to run, or can you provide an MC package, thank you.

You have to rewrite the dataset class by yourself, the code I provided is based on my own server environment. I'm not able to provide the MC package since it is an internal library just for our server environment.

**Comment by pc-cp** [comment]
Very effective communication, lots of gains. 👍
